// overlayscrollbars-svelte@0.5.1/overlayscrollbars-svelte.es.js
import { OverlayScrollbars as F } from "overlayscrollbars";
function M() {
}
function S(e, t) {
  for (const n in t)
    e[n] = t[n];
  return e;
}
function W(e) {
  return e();
}
function H() {
  return /* @__PURE__ */ Object.create(null);
}
function E(e) {
  e.forEach(W);
}
function X(e) {
  return typeof e == "function";
}
function Y(e, t) {
  return e != e ? t == t : e !== t || e && typeof e == "object" || typeof e == "function";
}
function me(e) {
  return Object.keys(e).length === 0;
}
function Z(e, t, n, r) {
  if (e) {
    const o = $(e, t, n, r);
    return e[0](o);
  }
}
function $(e, t, n, r) {
  return e[1] && r ? S(n.ctx.slice(), e[1](r(t))) : n.ctx;
}
function ee(e, t, n, r) {
  if (e[2] && r) {
    const o = e[2](r(n));
    if (t.dirty === void 0)
      return o;
    if (typeof o == "object") {
      const l = [], i = Math.max(t.dirty.length, o.length);
      for (let a = 0; a < i; a += 1)
        l[a] = t.dirty[a] | o[a];
      return l;
    }
    return t.dirty | o;
  }
  return t.dirty;
}
function te(e, t, n, r, o, l) {
  if (o) {
    const i = $(t, n, r, l);
    e.p(i, o);
  }
}
function ne(e) {
  if (e.ctx.length > 32) {
    const t = [], n = e.ctx.length / 32;
    for (let r = 0; r < n; r++)
      t[r] = -1;
    return t;
  }
  return -1;
}
function he(e) {
  const t = {};
  for (const n in e)
    n[0] !== "$" && (t[n] = e[n]);
  return t;
}
function J(e, t) {
  const n = {};
  t = new Set(t);
  for (const r in e)
    !t.has(r) && r[0] !== "$" && (n[r] = e[r]);
  return n;
}
function R(e, t, n) {
  e.insertBefore(t, n || null);
}
function j(e) {
  e.parentNode && e.parentNode.removeChild(e);
}
function re(e) {
  return document.createElement(e);
}
function ye(e) {
  return document.createTextNode(e);
}
function ge() {
  return ye("");
}
function T(e, t, n) {
  n == null ? e.removeAttribute(t) : e.getAttribute(t) !== n && e.setAttribute(t, n);
}
function K(e, t) {
  const n = Object.getOwnPropertyDescriptors(e.__proto__);
  for (const r in t)
    t[r] == null ? e.removeAttribute(r) : r === "style" ? e.style.cssText = t[r] : r === "__value" ? e.value = e[r] = t[r] : n[r] && n[r].set ? e[r] = t[r] : T(e, r, t[r]);
}
function Q(e, t) {
  Object.keys(t).forEach((n) => {
    pe(e, n, t[n]);
  });
}
function pe(e, t, n) {
  t in e ? e[t] = typeof e[t] == "boolean" && n === "" ? !0 : n : T(e, t, n);
}
function be(e) {
  return Array.from(e.childNodes);
}
function ve(e, t, { bubbles: n = !1, cancelable: r = !1 } = {}) {
  const o = document.createEvent("CustomEvent");
  return o.initCustomEvent(e, n, r, t), o;
}
let v;
function b(e) {
  v = e;
}
function D() {
  if (!v)
    throw new Error("Function called outside component initialization");
  return v;
}
function ke(e) {
  D().$$.on_mount.push(e);
}
function Ee(e) {
  D().$$.after_update.push(e);
}
function Ie(e) {
  D().$$.on_destroy.push(e);
}
function Ae() {
  const e = D();
  return (t, n, { cancelable: r = !1 } = {}) => {
    const o = e.$$.callbacks[t];
    if (o) {
      const l = ve(t, n, { cancelable: r });
      return o.slice().forEach((i) => {
        i.call(e, l);
      }), !l.defaultPrevented;
    }
    return !0;
  };
}
const p = [], w = [], O = [], V = [], Oe = Promise.resolve();
let x = !1;
function Ce() {
  x || (x = !0, Oe.then(oe));
}
function B(e) {
  O.push(e);
}
const q = /* @__PURE__ */ new Set();
let A = 0;
function oe() {
  const e = v;
  do {
    for (; A < p.length; ) {
      const t = p[A];
      A++, b(t), Se(t.$$);
    }
    for (b(null), p.length = 0, A = 0; w.length; )
      w.pop()();
    for (let t = 0; t < O.length; t += 1) {
      const n = O[t];
      q.has(n) || (q.add(n), n());
    }
    O.length = 0;
  } while (p.length);
  for (; V.length; )
    V.pop()();
  x = !1, q.clear(), b(e);
}
function Se(e) {
  if (e.fragment !== null) {
    e.update(), E(e.before_update);
    const t = e.dirty;
    e.dirty = [-1], e.fragment && e.fragment.p(e.ctx, t), e.after_update.forEach(B);
  }
}
const C = /* @__PURE__ */ new Set();
let h;
function we() {
  h = {
    r: 0,
    c: [],
    p: h
    // parent group
  };
}
function je() {
  h.r || E(h.c), h = h.p;
}
function g(e, t) {
  e && e.i && (C.delete(e), e.i(t));
}
function k(e, t, n, r) {
  if (e && e.o) {
    if (C.has(e))
      return;
    C.add(e), h.c.push(() => {
      C.delete(e), r && (n && e.d(1), r());
    }), e.o(t);
  } else
    r && r();
}
function De(e, t) {
  const n = {}, r = {}, o = { $$scope: 1 };
  let l = e.length;
  for (; l--; ) {
    const i = e[l], a = t[l];
    if (a) {
      for (const f in i)
        f in a || (r[f] = 1);
      for (const f in a)
        o[f] || (n[f] = a[f], o[f] = 1);
      e[l] = a;
    } else
      for (const f in i)
        o[f] = 1;
  }
  for (const i in r)
    i in n || (n[i] = void 0);
  return n;
}
function ze(e, t, n, r) {
  const { fragment: o, after_update: l } = e.$$;
  o && o.m(t, n), r || B(() => {
    const i = e.$$.on_mount.map(W).filter(X);
    e.$$.on_destroy ? e.$$.on_destroy.push(...i) : E(i), e.$$.on_mount = [];
  }), l.forEach(B);
}
function Ne(e, t) {
  const n = e.$$;
  n.fragment !== null && (E(n.on_destroy), n.fragment && n.fragment.d(t), n.on_destroy = n.fragment = null, n.ctx = []);
}
function Fe(e, t) {
  e.$$.dirty[0] === -1 && (p.push(e), Ce(), e.$$.dirty.fill(0)), e.$$.dirty[t / 31 | 0] |= 1 << t % 31;
}
function qe(e, t, n, r, o, l, i, a = [-1]) {
  const f = v;
  b(e);
  const s = e.$$ = {
    fragment: null,
    ctx: [],
    // state
    props: l,
    update: M,
    not_equal: o,
    bound: H(),
    // lifecycle
    on_mount: [],
    on_destroy: [],
    on_disconnect: [],
    before_update: [],
    after_update: [],
    context: new Map(t.context || (f ? f.$$.context : [])),
    // everything else
    callbacks: H(),
    dirty: a,
    skip_bound: !1,
    root: t.target || f.$$.root
  };
  i && i(s.root);
  let u = !1;
  if (s.ctx = n ? n(e, t.props || {}, (c, _, ...m) => {
    const y = m.length ? m[0] : _;
    return s.ctx && o(s.ctx[c], s.ctx[c] = y) && (!s.skip_bound && s.bound[c] && s.bound[c](y), u && Fe(e, c)), _;
  }) : [], s.update(), u = !0, E(s.before_update), s.fragment = r ? r(s.ctx) : !1, t.target) {
    if (t.hydrate) {
      const c = be(t.target);
      s.fragment && s.fragment.l(c), c.forEach(j);
    } else
      s.fragment && s.fragment.c();
    t.intro && g(e.$$.fragment), ze(e, t.target, t.anchor, t.customElement), oe();
  }
  b(f);
}
class Pe {
  $destroy() {
    Ne(this, 1), this.$destroy = M;
  }
  $on(t, n) {
    if (!X(n))
      return M;
    const r = this.$$.callbacks[t] || (this.$$.callbacks[t] = []);
    return r.push(n), () => {
      const o = r.indexOf(n);
      o !== -1 && r.splice(o, 1);
    };
  }
  $set(t) {
    this.$$set && !me(t) && (this.$$.skip_bound = !0, this.$$set(t), this.$$.skip_bound = !1);
  }
}
const Me = () => {
  if (typeof window > "u") {
    const s = () => {
    };
    return [s, s];
  }
  let e, t;
  const n = window, r = typeof n.requestIdleCallback == "function", o = n.requestAnimationFrame, l = n.cancelAnimationFrame, i = r ? n.requestIdleCallback : o, a = r ? n.cancelIdleCallback : l, f = () => {
    a(e), l(t);
  };
  return [
    (s, u) => {
      f(), e = i(
        r ? () => {
          f(), t = o(s);
        } : s,
        typeof u == "object" ? u : { timeout: 2233 }
      );
    },
    f
  ];
};
function xe(e) {
  let t;
  const n = (
    /*#slots*/
    e[13].default
  ), r = Z(
    n,
    e,
    /*$$scope*/
    e[12],
    null
  );
  return {
    c() {
      r && r.c();
    },
    m(o, l) {
      r && r.m(o, l), t = !0;
    },
    p(o, l) {
      r && r.p && (!t || l & /*$$scope*/
      4096) && te(
        r,
        n,
        o,
        /*$$scope*/
        o[12],
        t ? ee(
          n,
          /*$$scope*/
          o[12],
          l,
          null
        ) : ne(
          /*$$scope*/
          o[12]
        ),
        null
      );
    },
    i(o) {
      t || (g(r, o), t = !0);
    },
    o(o) {
      k(r, o), t = !1;
    },
    d(o) {
      r && r.d(o);
    }
  };
}
function Be(e) {
  let t, n;
  const r = (
    /*#slots*/
    e[13].default
  ), o = Z(
    r,
    e,
    /*$$scope*/
    e[12],
    null
  );
  return {
    c() {
      t = re("div"), o && o.c(), T(t, "data-overlayscrollbars-contents", "");
    },
    m(l, i) {
      R(l, t, i), o && o.m(t, null), e[14](t), n = !0;
    },
    p(l, i) {
      o && o.p && (!n || i & /*$$scope*/
      4096) && te(
        o,
        r,
        l,
        /*$$scope*/
        l[12],
        n ? ee(
          r,
          /*$$scope*/
          l[12],
          i,
          null
        ) : ne(
          /*$$scope*/
          l[12]
        ),
        null
      );
    },
    i(l) {
      n || (g(o, l), n = !0);
    },
    o(l) {
      k(o, l), n = !1;
    },
    d(l) {
      l && j(t), o && o.d(l), e[14](null);
    }
  };
}
function P(e) {
  let t, n, r, o;
  const l = [Be, xe], i = [];
  function a(u, c) {
    return (
      /*hydrated*/
      u[3] ? 0 : 1
    );
  }
  n = a(e), r = i[n] = l[n](e);
  let f = [
    { "data-overlayscrollbars-initialize": "" },
    /*$$restProps*/
    e[4]
  ], s = {};
  for (let u = 0; u < f.length; u += 1)
    s = S(s, f[u]);
  return {
    c() {
      t = re(
        /*element*/
        e[0]
      ), r.c(), /-/.test(
        /*element*/
        e[0]
      ) ? Q(t, s) : K(t, s);
    },
    m(u, c) {
      R(u, t, c), i[n].m(t, null), e[15](t), o = !0;
    },
    p(u, c) {
      let _ = n;
      n = a(u), n === _ ? i[n].p(u, c) : (we(), k(i[_], 1, 1, () => {
        i[_] = null;
      }), je(), r = i[n], r ? r.p(u, c) : (r = i[n] = l[n](u), r.c()), g(r, 1), r.m(t, null)), s = De(f, [
        { "data-overlayscrollbars-initialize": "" },
        c & /*$$restProps*/
        16 && /*$$restProps*/
        u[4]
      ]), /-/.test(
        /*element*/
        u[0]
      ) ? Q(t, s) : K(t, s);
    },
    i(u) {
      o || (g(r), o = !0);
    },
    o(u) {
      k(r), o = !1;
    },
    d(u) {
      u && j(t), i[n].d(), e[15](null);
    }
  };
}
function Re(e) {
  let t = (
    /*element*/
    e[0]
  ), n, r, o = (
    /*element*/
    e[0] && P(e)
  );
  return {
    c() {
      o && o.c(), n = ge();
    },
    m(l, i) {
      o && o.m(l, i), R(l, n, i), r = !0;
    },
    p(l, [i]) {
      /*element*/
      l[0] ? t ? Y(
        t,
        /*element*/
        l[0]
      ) ? (o.d(1), o = P(l), o.c(), o.m(n.parentNode, n)) : o.p(l, i) : (o = P(l), o.c(), o.m(n.parentNode, n)) : t && (o.d(1), o = null), t = /*element*/
      l[0];
    },
    i(l) {
      r || (g(o), r = !0);
    },
    o(l) {
      k(o), r = !1;
    },
    d(l) {
      l && j(n), o && o.d(l);
    }
  };
}
function Te(e, t, n) {
  const r = ["element", "options", "events", "defer", "osInstance", "getElement"];
  let o = J(t, r), { $$slots: l = {}, $$scope: i } = t, { element: a = "div" } = t, { options: f = void 0 } = t, { events: s = void 0 } = t, { defer: u = void 0 } = t, c = null, _ = null, m = null, y = !1, I, U;
  const [le, ie] = Me(), se = () => {
    if (!y)
      return;
    const d = () => {
      c == null || c.destroy(), n(10, c = F(
        {
          target: _,
          elements: { viewport: m, content: m }
        },
        f || {},
        I || {}
      ));
    };
    u ? le(d, u) : d(), U = a;
  }, L = {
    initialized: "osInitialized",
    updated: "osUpdated",
    destroyed: "osDestroyed",
    scroll: "osScroll"
  }, ue = Ae(), ce = () => c, fe = () => _;
  ke(() => {
    n(3, y = !0);
  }), Ie(() => {
    ie(), c == null || c.destroy();
  }), Ee(() => {
    U !== a && se();
  });
  function ae(d) {
    w[d ? "unshift" : "push"](() => {
      m = d, n(2, m);
    });
  }
  function de(d) {
    w[d ? "unshift" : "push"](() => {
      _ = d, n(1, _);
    });
  }
  return e.$$set = (d) => {
    t = S(S({}, t), he(d)), n(4, o = J(t, r)), "element" in d && n(0, a = d.element), "options" in d && n(5, f = d.options), "events" in d && n(6, s = d.events), "defer" in d && n(7, u = d.defer), "$$scope" in d && n(12, i = d.$$scope);
  }, e.$$.update = () => {
    if (e.$$.dirty & /*events*/
    64) {
      const d = s || {};
      n(11, I = Object.keys(L).reduce(
        (G, z) => {
          const N = d[z];
          return G[z] = [
            (..._e) => ue(
              // @ts-ignore
              L[z],
              // @ts-ignore
              _e
            ),
            ...(Array.isArray(N) ? N : [N]).filter(Boolean)
          ], G;
        },
        {}
      ));
    }
    e.$$.dirty & /*instance, options*/
    1056 && F.valid(c) && c.options(f || {}, !0), e.$$.dirty & /*instance, combinedEvents*/
    3072 && F.valid(c) && c.on(
      /* c8 ignore next */
      I || {},
      !0
    );
  }, [
    a,
    _,
    m,
    y,
    o,
    f,
    s,
    u,
    ce,
    fe,
    c,
    I,
    i,
    l,
    ae,
    de
  ];
}
class Ue extends Pe {
  constructor(t) {
    super(), qe(this, t, Te, Re, Y, {
      element: 0,
      options: 5,
      events: 6,
      defer: 7,
      osInstance: 8,
      getElement: 9
    });
  }
  get osInstance() {
    return this.$$.ctx[8];
  }
  get getElement() {
    return this.$$.ctx[9];
  }
}
const Ge = Ue;
export {
  Ge as OverlayScrollbarsComponent
};
